### DISCUSSION TITLE
Configuring Texlab for Neovim on macOS

### DISCUSSION THREAD

**Discussion by dpo** [latex-lsp/texlab#795] [Q&A]
Hello. I'm using Texlab with LunarVim and having trouble configuring it. I asked a question [over at LunarVim](https://github.com/LunarVim/LunarVim/discussions/3277), but haven't received any response.

My tentative configuration is as follows:
```lua
vim.list_extend(lvim.lsp.automatic_configuration.skipped_servers, { "texlab" })

require('lspconfig').texlab.setup({
  setting = {
    texlab = {
      forwardSearch = {
        executable = '/Applications/Skim.app/Contents/SharedSupport/displayline',
        args = { "-b", "%l", "%p", "%f" },
      },
      build = {
        onSave = true,
        forwardSearchAfter = true,
      },
      chktex = {
        onOpenAndSave = true,
      },
      formatterLineLength = 0,
    },
  },
})
```
Calling `:TexlabBuild` directly works, but `:TexlabForward` fails with the message `Search Unconfigured` (though I think I configured it above).

In addition, build on save isn't working (i.e., saving doesn't trigger a build).

Is there something visibly wrong with the configuration above? If not, how does one debug this kind of issue (I'm new to Neovim and Lua)?

Thanks!

**Comment by dpo** [comment]
Just pinging here in case someone has an idea. Still unable to use Texlab.

**Reply by pfoerster** [reply]
Hey, can you try creating a log file using `texlab -vvvv --log-file=/tmp/texlab.log`?
It looks like your client does not pass the LSP config to texlab. Building should work out of the box (it uses `latexmk` by default) but forward search requires extra configuration.

**Reply by dpo** [reply]
Do you mean that I have to have Neovim somehow fire up Texlab with those arguments? I have to investigate how to do that. If I simply write the command in a terminal, and then open Neovim and build, the log file remains empty.

**Reply by pfoerster** [reply]
Yeah, `Neovim` should pass those arguments to `texlab` or you can try and see if `Neovim` logs the JSON-RPC messages exchanged between the server and the client.

**Reply by dpo** [reply]
Ok I managed to get the server to start with the command you gave above (though I had to replace `=` with a space, right?). Indeed, it looks like the settings are not passed to texlab. Thank you.

**Reply by dpo** [reply]
Here's what I see in the log file:
```
DEBUG - < {"jsonrpc":"2.0","params":{"settings":{"texlab":{"build":{"onSave":false,"executable":"latexmk","forwardSearchAfter":false,"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"]},"forwardSearch":{"args":[]},"auxDirectory":".","diagnosticsDelay":300,"latexFormatter":"latexindent","chktex":{"onEdit":false,"onOpenAndSave":false},"bibtexFormatter":"texlab","formatterLineLength":80,"latexindent":{"modifyLineBreaks":false}}}},"method":"workspace\/didChangeConfiguration"}
```
